lia-ui/src/main/java/com/lithium/community/android/ui/components/adapters/LiConversationAdapter.java

```diff
@@ -169,136 +185,317 @@ private void showPopupMenu(final View view, final int position) {
+                    @Override
+                    public void onClick(final DialogInterface dialog, int which) {
+                        switch (which) {
+                            case DialogInterface.BUTTON_POSITIVE:
+                                deleteMessage(item, isParentItem);
+                                dialog.dismiss();
+                                break;
+
+                            case DialogInterface.BUTTON_NEGATIVE:
+                                dialog.dismiss();
+                                break;
+                            default:
+                                break;
                         }
+                    }
+                };
+                AlertDialog.Builder builder = new AlertDialog.Builder(activity);
+                builder.setMessage(R.string.li_are_you_sure)
+                        .setPositiveButton(R.string.li_yes, dialogClickListener)
+                        .setNegativeButton(R.string.li_no, dialogClickListener).show();
+
+                return true;
+            } else if (menuItem.getItemId() == R.id.li_action_subscribe) {
+                boolean isSubscribed = item.getUserContext().isSubscribed();
+                if (isSubscribed) {
+                    unsubscribe(item, position);
+                } else {
+                    subscribe(item, position);
+                }
+                return false;
+            } else {
+                return false;
+            }
+        }
+    }
 
-                        LiClientRequestParams.LiMarkMessageParams params = new LiClientRequestParams.LiMarkMessageParams(activity,
-                                String.valueOf(item.getAuthor().getId()),
-                                String.valueOf(item.getId()), isMarkUnread);
-                        try {
-                            LiClientManager.getMarkMessagePostClient(params).processAsync(new LiAsyncRequestCallback<LiPostClientResponse>() {
-                                @Override
-                                public void onSuccess(LiBaseRestRequest request, LiPostClientResponse response) throws LiRestResponseException {
-                                    if (response.getHttpCode() == LiCoreSDKConstants.HTTP_CODE_SUCCESSFUL) {
-                                        item.getUserContext().setRead(!item.getUserContext().getRead());
-                                        LiUIUtils.showMarkReadSuccessful(activity, item);
-                                    } else {
-                                        LiUIUtils.showMarkReadUnSuccessful(activity, item);
-                                    }
-                                }
+    private void openReportAbuse(LiMessage item) {
+        Bundle bundle = new Bundle();
+        bundle.putString(LiSDKConstants.REPORT_ABUSE_MESSAGE_ID, item.getId() + "");
+        bundle.putString(LiSDKConstants.REPORT_ABUSE_AUTHOR_ID, item.getAuthor().getId() + "");
```

Comment:
How about String.valueOf() ?

Reply:
this could introduce runtime exceptions, just in case if any of these happens to be null.